test/test-readability.js

```diff
@@ -47,11 +48,44 @@ var testPages = fs.readdirSync(testPageRoot).map(function(dir) {
   };
 });
 
+function suite(result, expectedContent, expectedMetadata) {
+  it("should return a result object", function() {
+    expect(result).to.include.keys("content", "title", "excerpt", "byline");
+  });
+
+  it("should extract expected content with jsdom", function() {
+    expect(expectedContent).eql(prettyPrint(result.content));
+  });
+
+  it("should extract expected title", function() {
+    expect(expectedMetadata.title).eql(result.title);
+  });
+
+  it("should extract expected byline", function() {
+    expect(expectedMetadata.byline).eql(result.byline);
+  });
+
+  it("should extract expected excerpt", function() {
+    expect(expectedMetadata.excerpt).eql(result.excerpt);
+  });
+}
+
+function removeCommentNodesRecursively(node) {
+  for (var n = 0, length = node.childNodes.length; n < length; n++) {
+    var child = node.childNodes[n];
+    if (!child) continue; // JSDOM is dumb sometimes :/
```

Comment:
> Can you elaborate? :-)
For some reason, jsdom retrieves a bunch of `undefined` items in `childNodes` list for no obvious reason.
> I'm sure they'll want to know.

They don't. The 3.x branch of jsdom is claimed [not officially maintained anymore since they moved to the io.js platform for 4.x](https://github.com/tmpvar/jsdom/tree/3.x#jsdom-3x-branch).

But you're invited to fork and maintain your own version if you will (hint: don't).